model_analyzer/analyzer.py

```diff
@@ -212,8 +213,18 @@ def _create_model_manager(self, client, gpus):
             metrics_manager=self._metrics_manager,
             state_manager=self._state_manager)
 
-    def _get_server_only_metrics(self, client):
+    def _get_server_only_metrics(self, client, gpus):
         if self._config.triton_launch_mode != 'c_api':
+            if self._do_checkpoint_gpus_match(gpus):
+                logger.info(
+                    "GPU devices match checkpoint - skipping server metric acquisition"
+                )
+                return
+            elif gpus is not None:
+                sys.exit(
```

Comment:
why sys.exit instead of raising an exception?

Reply:
I've changed this to be an exception